numpy/_core/_ufunc_config.py

```diff
@@ -167,6 +167,18 @@ def setbufsize(size):
     size : int
         Size of buffer.
 
+    Returns
+    -------
+    bufsize : int
+        Previous size of ufunc buffer in bytes.
+
+    Examples
+    --------
+    >>> np.setbufsize(4096)
+    8192
+    >>> np.getbufsize()
+    4096
```

Comment:
I think it would be better to show this while using an errstate context manager, to show how the bufsize will be reset once the errorstate is exited:
```
>>> with np.errstate():
...     np.setbufsize(4096)
...     print(np.getbufsize())
... 
8192
4096
>>> np.getbufsize()
8192
>>> 
```

Reply:
I added the example with a small description before.

If using this context manager is the expected usage of this function I can remove the first example. If not, I think it should have both examples.

Either way I can remove the first example if you think it will be clearer for readers.

Reply:
When running with `errstate` `setbufsize` returns the new value somehow. I'm going to investigate why that happens.

```
=====
numpy
=====

numpy.setbufsize
----------------

File "venv/lib/python3.11/site-packages/numpy/__init__.py", line 184, in setbufsize
Failed example:
    with np.errstate():
        np.setbufsize(4096)
        print(np.getbufsize())
Expected:
    8192
    4096
Got:
    4096
    4096
```